### PR TITLE
fix(storefront): BCTHEME-207 fix focus on sort by

### PR DESCRIPTION
**Pull request by bc-alexsaiannyi**
#### What?

This PR addresses focus issue on SortBy element after reloading page. It can be seen as a solution before implementing partly page update mechanism. At the moment, new method `arrangeFocusOnSortBy` has been added to CatalogPage Class and can be used on pages where sortBy select is presented (such as category or search). CatalogPage constructor helps to set up event listener that tracks if sortBy is active and keeps it on LocalStorage  in that case.

#### Tickets / Documentation
- [BCTHEME-207](https://jira.bigcommerce.com/browse/BCTHEME-207)

#### Screenshots (if appropriate)
https://user-images.githubusercontent.com/67792608/105361889-dbaa6780-5c02-11eb-8d58-c2bdc5bdf070.mov

### CODE REVIEW

File: assets/js/theme/category.js [hunk 1]
```diff
@@ -8,16 +8,33 @@ export default class Category extends CatalogPage {
...
+
+    arrangeFocusOnElement($element) {
+        if (this.isSortBySelected) {
+            $element.focus();
+            window.localStorage.removeItem('sortByStatus');
+        }
     }
 
     onReady() {
+        const sortBySelector = $('[data-sort-by="product"] #sort');
```
Comment (BC-tymurbiedukhin) [inline assets/js/theme/category.js:26]: Could we use only #sort id here?

**Review by yurytut1993 — APPROVED** [review]

File: assets/js/theme/catalog.js [hunk 2]
```diff
@@ -3,6 +3,23 @@ import urlUtils from './common/utils/url-utils';
...
+        super(context);
+
+        window.addEventListener('beforeunload', () => {
+            if (document.activeElement.id === 'sort') {
+                window.localStorage.setItem('sortByStatus', 'selected');
+            }
+        });
+    }
+
+    arrangeFocusOnSortBy($elementSelector) {
```
Comment (BC-tymurbiedukhin) [inline assets/js/theme/catalog.js:16]: I believe we could skip passing selector here

Comment (bc-alexsaiannyi) [reply assets/js/theme/catalog.js]: I've added it in case the sortBy Selector will be change later for some reason. Removed it.

**Review by BC-tymurbiedukhin — APPROVED** [review]
LGTM

**Pull request merged**